File: framework/languages/java/samples/java/GameQuest/Server/Configuration/src/main/java/systems/zlink/samples/gamequest/server/configuration/QuestMissionInstanceTopology.java
package systems.zlink.samples.gamequest.server.configuration;

import systems.zlink.contracts.core.RoutingId;

/** Resolved endpoints and owner index for a single QuestMission instance. */
public record QuestMissionInstanceTopology(
    String missionName,
    String spotEndpoint,
    String spotRouterEndpoint,
    RoutingId spotRid,
    int ownerIndex) {
}

File: framework/languages/java/samples/kotlin/GameQuest/Server/Configuration/src/main/kotlin/systems/zlink/samples/kotlin/gamequest/server/configuration/QuestMissionInstanceTopology.kt
package systems.zlink.samples.kotlin.gamequest.server.configuration

import systems.zlink.contracts.core.RoutingId

/** Resolved endpoints and owner index for a single QuestMission instance. */
data class QuestMissionInstanceTopology(
    val missionName: String,
    val spotEndpoint: String,
    val spotRouterEndpoint: String,
    val spotRid: RoutingId,
    val ownerIndex: Int,
)
